ports/stm32/qspi.c

```diff
@@ -74,11 +78,33 @@ static inline void qspi_mpu_enable_mapped(void) {
     // for the memory-mapped region, so 3 MPU regions are used to disable access
     // to everything except the valid address space, using holes in the bottom
     // of the regions and nesting them.
-    // At the moment this is hard-coded to 2MiB of QSPI address space.
     uint32_t irq_state = mpu_config_start();
+    #if MICROPY_HW_QSPI_MPU_REGION_SIZE > 128
+    mpu_config_region(MPU_REGION_QSPI1, QSPI_MAP_ADDR, MPU_CONFIG_DISABLE(0xFF, MPU_REGION_SIZE_256MB));
```

Comment:
For this case I think just do nothing, because 0xff means it's disabling the MPU for the entire 256MB region.  Hence just remove this line (and maybe add a comment that the region max size is 256MB).

Reply:
But `qspi_mpu_disable_all` from `ioctl` disables access to the entire region when the bus is acquired, shouldn't `qspi_memory_map` re-enable the access to the region ?

Reply:
> But `qspi_mpu_disable_all` from `ioctl` disables access to the entire region when the bus is acquired, shouldn't `qspi_memory_map` re-enable the access to the region ?

You are right.  So this line is needed, to re-enable access.

> > 0xff means it's disabling the MPU for the entire 256MB region.
> 
> It's not, it's enabling MPU access for the entire region, since there's no other region that overlaps the disabled (0xff) range.

It's masking the MPU (ie disabling the MPU from preventing memory acces), and therefore enabling memory accesses.  A "1" bit in the first arg to MPU_CONFIG_DISABLE means the MPU is not active in that chunk, and a "0" bit means the MPU is active in that chunk and prevents memory accesses.

Reply:
Yes more or less. I find the description of SRD in the app notes/manual is confusing, it says:

> 1 Corresponding sub-region is disabled.

It's more clearly stated in the "ARM Cortex-M7 Devices Generic User Guide" document:

> Disabling a subregion means another region overlapping the disabled range matches instead. If no other enabled region overlaps the disabled subregion, and the access is unprivileged or the background region is
disabled, the MPU issues a fault.

So either the privileged access or the background region that makes this work. I think the above should be a comment in the code.

Reply:
If you want to add a comment, feel free.

Reply:
Done.